File: src/main/java/com/dsb/rest/controller/PanopliesController.java
package com.dsb.rest.controller;

import com.dsb.rest.model.Panoplie;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.io.FileReader;
import java.io.IOException;
import java.util.Arrays;
import java.util.List;

@RestController
@CrossOrigin(origins = {"http://localhost:4200", "https://testdsb.herokuapp.com"})
@RequestMapping(path = "/panoplies")
public class PanopliesController {
    private final static String dir = "src/main/resources/panoplies.json";

    @GetMapping(path = "/all", produces = MediaType.APPLICATION_JSON_VALUE)
    public List<Panoplie> getAllPanoplies() throws IOException {
        ObjectMapper mapper = new ObjectMapper();
        return Arrays.asList(mapper.readValue(new FileReader(dir), Panoplie[].class));
    }
}

File: src/main/kotlin/com/dsb/rest/controller/PanopliesController.kt
package com.dsb.rest.controller

import com.dsb.rest.dao.PanopliesDAO
import com.dsb.rest.model.Panoplie
import com.dsb.rest.model.Statistic
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.io.IOException

@RestController
@RequestMapping(path = ["/panoplies"])
class PanopliesController(private val panopliesDAO: PanopliesDAO) {
    @Throws(
        IOException::class
    )
    @GetMapping(path = ["/all"], produces = [MediaType.APPLICATION_JSON_VALUE])
    fun getAllPanoplies(): List<Panoplie> = panopliesDAO.getAllPanoplies()

    @GetMapping(path = ["/"], produces = [MediaType.APPLICATION_JSON_VALUE])
    @Throws(
        IOException::class
    )
    fun getPanoplie(@RequestParam id: Int, @RequestParam numberOfItemInPanoplie: Int): List<Statistic> {
        return panopliesDAO.getStatsOfPanoplieBonus(id, numberOfItemInPanoplie)
    }
}